pkg/plugins/runtime/k8s/webhooks/validation.go

```diff
@@ -0,0 +1,63 @@
+package webhooks
+
+import (
+	"context"
+	"net/http"
+
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	core_model "github.com/Kong/kuma/pkg/core/resources/model"
+	core_registry "github.com/Kong/kuma/pkg/core/resources/registry"
+	k8s_resources "github.com/Kong/kuma/pkg/plugins/resources/k8s"
+	k8s_model "github.com/Kong/kuma/pkg/plugins/resources/k8s/native/pkg/model"
+	k8s_registry "github.com/Kong/kuma/pkg/plugins/resources/k8s/native/pkg/registry"
+)
+
+func NewValidatingWebhook(converter k8s_resources.Converter, coreRegistry core_registry.TypeRegistry, k8sRegistry k8s_registry.TypeRegistry) (*admission.Webhook, error) {
+	return &admission.Webhook{
+		Handler: &validatingHandler{
+			coreRegistry: coreRegistry,
+			k8sRegistry:  k8sRegistry,
+			converter:    converter,
+		},
+	}, nil
+}
+
+type validatingHandler struct {
+	coreRegistry core_registry.TypeRegistry
+	k8sRegistry  k8s_registry.TypeRegistry
+	converter    k8s_resources.Converter
+	decoder      *admission.Decoder
+}
+
+func (h *validatingHandler) InjectDecoder(d *admission.Decoder) error {
+	h.decoder = d
+	return nil
+}
+
+func (h *validatingHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	resType := core_model.ResourceType(req.Kind.Kind)
```

Comment:
We need a way to guarantee this assumption (that `k8s kind == resource type` ).

E.g., a dedicated unit test to prevent accidental renaming